x/sync/network_server.go

```diff
@@ -138,11 +138,14 @@ func (s *NetworkServer) HandleChangeProofRequest(
 	requestID uint32,
 	req *pb.SyncGetChangeProofRequest,
 ) error {
+	if req.EndKey == nil {
```

Comment:
not sure if we have valid representation with `req.EndKey == nil`, why do we need `IsNothing`?